src/components/AggregateList.tsx

```diff
@@ -100,7 +115,38 @@ export default class AggregateList extends React.Component<AggregateListProps, A
 
   makeLibraryNameList(category: string): JSX.Element[] {
     let libraries = this.props.data[category];
-    return libraries.map(l => <li className="inner-stats-item" key={l.uuid}><p>{l.basic_info.name}</p></li>);
+    return libraries.map((l) => {
+      return (
+        <li className="inner-stats-item" key={l.uuid}><p>{l.basic_info.name}{this.getGeographicInfo(l.areas)}</p></li>
+      );
+    });
+  }
+
+  getGeographicInfo(areas: { [key: string]: string[] }): string {
+    if (!this.state.geographicInfo) {
+      return "";
+    }
+    let areaString: string = "";
+    let allAreas: string[] = areas.focus.concat(areas.service);
+    allAreas.map((a: string) => {
+      // Each string is in the format "Zip code/city, (state abbreviation)".  We're just interested in the state abbreviation
+      // right now, so we get it by pulling out any two-letter sequence between parentheses.
+      // (If there's something longer than two letters between parentheses, it's the server-generated string "unknown",
+      // which we don't need to add individually to areaString; if everything was unknown, we'll handle it later.)
+      let regExp: RegExp = /\(([^)]{2})\)/;
+      let match: string[] = regExp.exec(a);
+      if (!match) {
+        return;
```

Comment:
An empty string should be returned.

Reply:
Do you have a test for this? It should've thrown an error since you are explicitly returning a string type.

Reply:
Oh, I see, this is not in the function but the loop. I think `forEach` would be better so you don't create a new array every time `map` is called (there are a lot of libraries in production and this will run a lot).